File: etc/167. Two_Sum_II_Input_Array_Is_Sorted.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize) {
    int start, end;
    int *answer = (int *)malloc(sizeof(int)*2);
    start, end = 0, (numbersSize--);
    while(start<end){
        int sum = numbers[start] + numbers[end];
        if(sum==target){break;}
        if(sum>target){end--;}
        else{start++;}
    }
    answer[0] = start+1; 
    answer[1] = end+1;
    return answer;  
}

File: SH/etc/167. Two_Sum_II_Input_Array_Is_Sorted.py
def twoSum(numbers: List[int], target: int) -> List[int]:
        start_1, end_1 = 0, len(numbers)-1
        while(start_1<end_1):
            sum = numbers[start_1]+numbers[end_1]
            if sum == target:
                    return [start_1+1, end_1+1]
            if sum > target:
                end_1 -= 1
            if sum < target:
                start_1 += 1
        return []